hazelcast/src/main/java/com/hazelcast/map/impl/client/AbstractMapQueryRequest.java

```diff
@@ -65,7 +65,7 @@ public AbstractMapQueryRequest(String name, IterationType iterationType) {
 
     @Override
     protected final void invoke() {
-        QueryResultSet result = new QueryResultSet(null, iterationType, true);
+        List<QueryResultRow> result = new ArrayList<QueryResultRow>();
```

Comment:
Collection of a IdentifiedDataSerializable/Portable is not supported in other clients. We can't send this as response directly.